File: 05-may/12-minimum-initial-energy-to-finish-tasks/solution.java
import java.util.Arrays;

class Solution {
    public int minimumEffort(int[][] tasks) {
        Arrays.sort(tasks, (a, b) -> (b[1] - b[0]) - (a[1] - a[0]));

        int spent = 0;
        int answer = 0;

        for (int[] task : tasks) {
            int actual = task[0];
            int minimum = task[1];

            answer = Math.max(answer, spent + minimum);
            spent += actual;
        }

        return answer;
    }
}

File: 05-may/12-minimum-initial-energy-to-finish-tasks/solution.py
from typing import List


class Solution:
    def minimumEffort(self, tasks: List[List[int]]) -> int:
        tasks.sort(key=lambda task: task[1] - task[0], reverse=True)

        spent = 0
        answer = 0

        for actual, minimum in tasks:
            answer = max(answer, spent + minimum)
            spent += actual

        return answer
